### PR TITLE
chore(CPlusPlus): delete alternate characters in a string

### PR DESCRIPTION
**Pull request by samughodake**
A simple C++ program on a string to delete alternate characters in a string

### CODE REVIEW

**Comment by samughodake** [comment]
Please any one can tell me why the PR run failed?
So that I can make corrections in it.

**Comment by ming-tsai** [comment]
> Please any one can tell me why the PR run failed?
> So that I can make corrections in it.

Hi @samughodake,
I assign to @Arsenic-ATG to review this pull request

File: algorithms/CPlusPlus/Strings/delete-alternate-characters.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+Description: To delete alternate characters in a string
+Approach: To iterate over string and append the alternate characters in the output string
+*/
+
+// Initial Template for C++
+#include <bits/stdc++.h>
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/delete-alternate-characters.cpp:7]: No need to clutter headers when it's just header that you want. (same in other file also)

```suggestion
#include <string>
```

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
+Description: A C++ program to print first letter of every word in a string
+*/
+
+
+//importing c++ libraries
+#include <bits/stdc++.h>
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/print-alphabets.cpp:7]:
```suggestion
#include <string>
```

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//function
+string firstAlphabet(string s){
+  //declaring an output string 'o'
+string o;
+//inserting 1st letter in the output string
+	  o+=s[0];
+    //iterating over the string from index 1
+    //if space is found, then the next character is appended in the output string
+	  for(int i=1;i<s.size();i++){
+	      if(s[i]==' '){
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/print-alphabets.cpp:20]: what if there are more than one space between 2 words ?

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
apart form the stuff mentioned in comments, also update the appropriate index readme.md with your creation in it.

File: algorithms/CPlusPlus/Strings/delete-alternate-characters.cpp (hunk 1 above)
Comment (samughodake) [reply algorithms/CPlusPlus/Strings/delete-alternate-characters.cpp]: Changes done!

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp (hunk 2 above)
Comment (samughodake) [reply algorithms/CPlusPlus/Strings/print-alphabets.cpp]: changes made

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp (hunk 3 above)
Comment (samughodake) [reply algorithms/CPlusPlus/Strings/print-alphabets.cpp]: updated the code for that too

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+string firstAlphabet(string s){
+  //declaring an output string 'o'
+string o;
+//inserting 1st letter in the output string if its not space
+if(s[0]!= ' '){
+   o+=s[0];  
+}
+	 
+    //iterating over the string from index 1
+    //if space is found, then we will keep checking if there are mulitple spaces between 2 words
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Strings/print-alphabets.cpp:21]: minor typo detected by codespell
```suggestion
    //if space is found, then we will keep checking if there are multiple spaces between 2 words
```

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
The codes look fine to me but there are some weird formatting issue I am seeing in there, it might be because of mixing the use of whitespace and tabs for indentation.

But that can be easily fixed with automated tools like clang format or [this] online formatter.

File: algorithms/CPlusPlus/Strings/print-alphabets.cpp (hunk 4 above)
Comment (samughodake) [reply algorithms/CPlusPlus/Strings/print-alphabets.cpp]: Hi, I made the following changes
1. fixed typo error
2. fixed indentation

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
Nice, now all that is left is to update the [appropriate readme.md](https://github.com/samughodake/DSA/blob/cpp-program-on-string/algorithms/CPlusPlus/README.md) with your program in it.

**Comment by samughodake** [comment]
> Nice, now all that is left is to update the [appropriate readme.md](https://github.com/samughodake/DSA/blob/cpp-program-on-string/algorithms/CPlusPlus/README.md) with your program in it.

Hi,
How to update readme.md file and programs? (I am very new to open source contribution)
I want to actively contribute to this repository.

**Comment by Arsenic-ATG** [comment]
> > Nice, now all that is left is to update the [appropriate readme.md](https://github.com/samughodake/DSA/blob/cpp-program-on-string/algorithms/CPlusPlus/README.md) with your program in it.
> 
> Hi,
> How to update readme.md file and programs? (I am very new to open source contribution)
> I want to actively contribute to this repository.

Every language directory of the repo have a [Readme.md](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/README.md) which is just the index of the all the programs we have implemented in the language.

All you have to do is add a link to your program in the appropriate section there, In your case it would be 2 programs under `String` category there 

You can learn more about it in the [contribution guidelines](https://github.com/MakeContributions/DSA#1-contribution-specifications) of the repo.

If you want, you can also have a look at some accepted PRs as an example.

**Comment by samughodake** [comment]
> > > Nice, now all that is left is to update the [appropriate readme.md](https://github.com/samughodake/DSA/blob/cpp-program-on-string/algorithms/CPlusPlus/README.md) with your program in it.
> > 
> > 
> > Hi,
> > How to update readme.md file and programs? (I am very new to open source contribution)
> > I want to actively contribute to this repository.
> 
> Every language directory of the repo have a [Readme.md](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/README.md) which is just the index of the all the programs we have implemented in the language.
> 
> All you have to do is add a link to your program in the appropriate section there, In your case it would be 2 programs under `String` category there
> 
> You can learn more about it in the [contribution guidelines](https://github.com/MakeContributions/DSA#1-contribution-specifications) of the repo.
> 
> If you want, you can also have a look at some accepted PRs as an example.

Thanks for this!

**Comment by samughodake** [comment]
Hi @Arsenic-ATG,
Why this PR is giving cpplint error? 
Please can you help me with these, so that I can correct it.

**Comment by Arsenic-ATG** [comment]
> Hi @Arsenic-ATG,
> Why this PR is giving cpplint error?
> Please can you help me with these, so that I can correct it.

The error says :
```
algorithms/CPlusPlus/Strings/print-alphabets.cpp:8:  Found C system header after C++ system header. Should be: print-alphabets.h, c system, c++ system, other.  [build/include_order] [4]
```

usually this is because of wrong [include order of headers](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) in a C++ program according to google style guide, but in your case it is due to `<bits/stdc++.h>`. 

Looks like, while force pushing the branch, you accidentally undid the changes pointed out in the pull request earlier. 

How did you updated your local branch ?

**Comment by samughodake** [comment]
> > Hi @Arsenic-ATG,
> > Why this PR is giving cpplint error?
> > Please can you help me with these, so that I can correct it.
> 
> The error says :
> 
> > algorithms/CPlusPlus/Strings/print-alphabets.cpp:8:  Found C system header after C++ system header. Should be: print-alphabets.h, c system, c++ system, other.  [build/include_order] [4]
> 
> usually this is because of wrong [include order of headers](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) in a C++ program according to google style guide, but in your case it is due to `<bits/stdc++.h>`.
> 
> Looks like, while force pushing the branch, you accidentally undid the changes pointed out in the pull request earlier.
> 
> How did you updated your local branch ?

Looks like I have messed up things a lot.
Please can you tell me how to fix things?

**Comment by Arsenic-ATG** [comment]
> > > Hi @Arsenic-ATG,
> > > Why this PR is giving cpplint error?
> > > Please can you help me with these, so that I can correct it.
> > 
> > 
> > The error says :
> > > algorithms/CPlusPlus/Strings/print-alphabets.cpp:8:  Found C system header after C++ system header. Should be: print-alphabets.h, c system, c++ system, other.  [build/include_order] [4]
> > 
> > 
> > usually this is because of wrong [include order of headers](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) in a C++ program according to google style guide, but in your case it is due to `<bits/stdc++.h>`.
> > Looks like, while force pushing the branch, you accidentally undid the changes pointed out in the pull request earlier.
> > How did you updated your local branch ?
> 
> Looks like I have messed up things a lot.
> Please can you tell me how to fix things?

What did you for which you had to force update the remote ? were you rebasing some commits ?

**Comment by samughodake** [comment]
> > > > Hi @Arsenic-ATG,
> > > > Why this PR is giving cpplint error?
> > > > Please can you help me with these, so that I can correct it.
> > > 
> > > 
> > > The error says :
> > > > algorithms/CPlusPlus/Strings/print-alphabets.cpp:8:  Found C system header after C++ system header. Should be: print-alphabets.h, c system, c++ system, other.  [build/include_order] [4]
> > > 
> > > 
> > > usually this is because of wrong [include order of headers](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) in a C++ program according to google style guide, but in your case it is due to `<bits/stdc++.h>`.
> > > Looks like, while force pushing the branch, you accidentally undid the changes pointed out in the pull request earlier.
> > > How did you updated your local branch ?
> > 
> > 
> > Looks like I have messed up things a lot.
> > Please can you tell me how to fix things?
> 
> What did you do for which you had to force update the remote ? were you rebasing some commits ? (Because if that's the case, then it shouldn't be hard to get those commits back)

Actually forgot to pull changes first and accidentally forced the commit.

**Comment by samughodake** [comment]
Will make another pull request for the above problem.

**Comment by Arsenic-ATG** [comment]
> Will make another pull request for the above problem on a different branch. This branch is actually conflicting with the readme file

That should be resolved now.
For now, just perform the changes I initially pointed out in the files on your branch and you should be good to go.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
Just fix the changes pointed out earlier ( marked as unresolved now ), similar to how you did before and the PR should be good to go.

**Comment by samughodake** [comment]
closing this pull request because of some git errors

**Comment by ming-tsai** [comment]
Hi @samughodake,
Wich error is appearing?

**Comment by Arsenic-ATG** [comment]
> Hi @samughodake,
> Wich error is appearing?

I don't know, The PR was all ok
although a force push before by the OP made them lose two commits, but those were very simple ones and they were only required to redo them again ( or a quick search in reflog of the local repo, if the commits were made locally )

**Comment by Arsenic-ATG** [comment]
> So, @samughodake only has to merge with `master` branch?

no, they already did that.

What I thought would have happened was OP's local repo ( on their system ) might be 2 commits before the remote fork and instead of pulling the changes and then pushing, they force pushed the repo instead, which lead them to loose those 2 commits. 

But the commits were very trivial changes and could have been easily recreated.

**Comment by ming-tsai** [comment]
> > So, @samughodake only has to merge with `master` branch?
> 
> no, they already did that.
> 
> What I thought would have happened was OP's local repo ( on their system ) might be 2 commits before the remote fork and instead of pulling the changes and then pushing, they force pushed the repo, which lead them to loose those 2 commits. 
> 
> But the commits were very trivial changes and could have been easily recreated.

Ok, so @samughodake will you open a new pull request for all changed or just reopen this one and commit all changed?

**Comment by samughodake** [comment]
> > > So, @samughodake only has to merge with `master` branch?
> > 
> > 
> > no, they already did that.
> > What I thought would have happened was OP's local repo ( on their system ) might be 2 commits before the remote fork and instead of pulling the changes and then pushing, they force pushed the repo, which lead them to loose those 2 commits.
> > But the commits were very trivial changes and could have been easily recreated.
> 
> Ok, so @samughodake will you open a new pull request for all changed or just reopen this one and commit all changed?

Sir @ming-tsai, I will open a new open pull request for this.

**Comment by ming-tsai** [comment]
Hi @samughodake,
This is pull request #469?

**Comment by samughodake** [comment]
> Hi @samughodake,
> Does this pull request #469 is the re-open one?

Yes, I closed this pull request and made a new one #480

**Comment by ming-tsai** [comment]
> > Hi @samughodake,
> > Does this pull request #469 is the re-open one?
> 
> Yes, I closed this pull request and made a new one #480 

Sorry, I put the incorrect pull request.

**Pull request closed without merging**